home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/RSQLNewsIT.java

```diff
@@ -0,0 +1,277 @@
+
+        assertFalse(listNews.isEmpty());
+    }
+
+    @Test
+    void getAllUsersDescSortBySourceTest() throws ApiException {
+        sort = "source,desc";
+        List<String> sourceList = new ArrayList<>();
+
+        List<ReadNews> listNews = newsApi
+                .getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source);
+        listNews.forEach(readUser -> sourceList.add(readUser.getSource()));
+        String[] actualSource = new String[sourceList.size()];
+        sourceList.toArray(actualSource);
+        Collections.sort(sourceList, Collections.reverseOrder());
+        String[] expectedSource = new String[sourceList.size()];
+        sourceList.toArray(expectedSource);
+
+        assertArrayEquals(expectedSource, actualSource);
+    }
+
+    @Test
+    void getAllNewsAscSortBySourceTest() throws ApiException {
+        sort = "source,asc";
+        List<String> sourceList = new ArrayList<>();
+
+        List<ReadNews> listNews = newsApi
+                .getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source);
+        listNews.forEach(readUser -> sourceList.add(readUser.getSource()));
+        String[] actualSource = new String[sourceList.size()];
+        sourceList.toArray(actualSource);
+        Collections.sort(sourceList);
+        String[] expectedSource = new String[sourceList.size()];
+        sourceList.toArray(expectedSource);
+
+        assertArrayEquals(expectedSource, actualSource);
+    }
+
+    @Test
+    void getAllUsersSortAndSearchTest() throws ApiException {
+        List<String> sourceList = new ArrayList<>();
+        sort = "source,asc";
+        filter = new StringBuilder()
+                .append("id=bt=('")
+                .append(newsList.get(1).getId())
+                .append("',")
+                .append("'")
+                .append(newsList.get(newsList.size() - 1).getId())
+                .append("')").toString();
+
+        List<ReadNews> listNews = newsApi
+                .getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source);
+        listNews.forEach(readUser -> sourceList.add(readUser.getSource()));
+        String[] actualSource = new String[sourceList.size()];
+        sourceList.toArray(actualSource);
+        Collections.sort(sourceList);
+        String[] expectedSource = new String[sourceList.size()];
+        sourceList.toArray(expectedSource);
+
+        assertEquals(listNews.size(), 4);
```

Comment:
How do we know the exact size? What if more than 4 exist in db? Sound like this test is dirty.